src/app/components/Header.tsx

```diff
@@ -1,60 +1,108 @@
-import React from "react";
-import Link from "next/link";
-import Image from "next/image";
-import { AppBar, Toolbar, Box, Button } from "@mui/material";
+"use client";  // Indicates that this is a client-side component
+
+import React, { useState } from 'react';
+import Link from 'next/link';
+import Image from 'next/image';
+import {AppBar,Toolbar,Box,Button,IconButton,Drawer,List,ListItem,ListItemButton,ListItemText} from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+import CloseIcon from '@mui/icons-material/Close';
 
 export default function Header() {
+  const [isDrawerOpen, setDrawerOpen] = useState(false);
```

Comment:
Follow the convention for useState
```suggestion
  const [isDrawerOpen, setIsDrawerOpen] = useState<boolean>(false);
```